### PR TITLE
Provide a help target for make

### PR DESCRIPTION
**Pull request by glyn**
I came across this on another project and wondered if it had any value here:
```
$ make help
all                            Build, test, and regenerate docs
clean                          Delete build output
build                          Build riff
test                           Run the tests
coverage                       Run the tests with coverage and race detection
docs                           Generate documentation
verify-docs                    Verify the generated docs are up to date
check-mockery                  Check that mockery is available and print installation instructions if it isn't
gen-mocks                      Generate mocks
help                           Print help for each make target

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -9,52 +9,56 @@ LDFLAGS_VERSION = -X github.com/projectriff/riff/pkg/cli.cli_name=riff \
...
-coverage:
+coverage: ## Run the tests with coverage and race detection
 	go test -mod=vendor -v --race -coverprofile=coverage.txt -covermode=atomic ./...
 
 $(OUTPUT): $(GO_SOURCES) VERSION
 	go build -mod=vendor -o $(OUTPUT) -ldflags "$(LDFLAGS_VERSION)" ./cmd/riff
 
 .PHONY: release
-release: $(GO_SOURCES) VERSION
+release: $(GO_SOURCES) VERSION $$ Cross-compile riff for various operating systems
```
Comment (scothis) [inline Makefile]:
```suggestion
release: $(GO_SOURCES) VERSION ## Cross-compile riff for various operating systems
```

File: Makefile [hunk 2]
```diff
@@ -9,52 +9,56 @@ LDFLAGS_VERSION = -X github.com/projectriff/riff/pkg/cli.cli_name=riff \
...
 	GOOS=darwin   GOARCH=amd64 go build -mod=vendor -ldflags "$(LDFLAGS_VERSION)" -o $(OUTPUT)     ./cmd/riff && tar -czf riff-darwin-amd64.tgz $(OUTPUT)     && rm -f $(OUTPUT)
 	GOOS=linux    GOARCH=amd64 go build -mod=vendor -ldflags "$(LDFLAGS_VERSION)" -o $(OUTPUT)     ./cmd/riff && tar -czf riff-linux-amd64.tgz  $(OUTPUT)     && rm -f $(OUTPUT)
 	GOOS=windows  GOARCH=amd64 go build -mod=vendor -ldflags "$(LDFLAGS_VERSION)" -o $(OUTPUT).exe ./cmd/riff && zip -mq riff-windows-amd64.zip $(OUTPUT).exe && rm -f $(OUTPUT).exe
 
-docs: $(OUTPUT) clean-docs
+docs: $(OUTPUT) clean-docs   ## Generate documentation
```
Comment (scothis) [inline Makefile]:
```suggestion
docs: $(OUTPUT) clean-docs ## Generate documentation
```

**Comment by glyn** [comment]
@ericbottard Scott likes this. If you like it too, please merge.

**Comment by ericbottard** [comment]
This serious dope has been merged as 5c81e052

**Pull request closed without merging**